backend/src/v5/schemas/tickets/templates.js

```diff
@@ -0,0 +1,152 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { defaultProperties, fieldTypes, presetEnumValues, presetModules, presetModulesProperties } = require('./templates.constants');
+const { types, utils: { stripWhen } } = require('../../utils/helper/yup');
+const Yup = require('yup');
+const { isString } = require('../../utils/helper/typeCheck');
+
+const typeNameToType = {
+	[fieldTypes.TEXT]: types.strings.title,
+	[fieldTypes.LONG_TEXT]: types.strings.longDescription,
+	[fieldTypes.BOOLEAN]: Yup.boolean(),
+	[fieldTypes.DATE]: types.date,
+	[fieldTypes.NUMBER]: Yup.number(),
+	[fieldTypes.ONE_OF]: types.strings.title,
+	[fieldTypes.MANY_OF]: Yup.array().of(types.strings.title),
+};
+
+const defaultFalse = stripWhen(Yup.boolean().default(false), (v) => !v);
+
+const fieldSchema = Yup.object().shape({
+	name: types.strings.title.required().min(1),
+	type: Yup.string().oneOf(Object.values(fieldTypes)).required(),
+	deprecated: defaultFalse,
+	required: defaultFalse,
+	values: Yup.mixed().when(['type'], (val, schema) => {
```

Comment:
`[type]` and `type` here is interchangable, but i'll remove it to avoid confusion.

type is always required yes, not sure what that has to do with this line though? we utilise `when` here to get access to the value in type, as values schema depends on what the type is. We never expect that to be undefined.

Reply:
docs of `when` here if it helps https://github.com/jquense/yup#schemawhenkeys-string--string-builder-object--values-any-schema--schema-schema